server/app/lib/actions/fusor/deployment/rhev/ose_launch.rb

```diff
@@ -65,6 +66,26 @@ def run
               #TODO save host.ip and host.name in deployment
             end
 
+            # Generate SSH Keys
+            key_type = 'rsa'
+            generate_ssh_keys(key_type)
+
+            # Distribute the key to each Master Nodes
+            for i in 1..deployment.openshift_number_master_nodes do
+              hostname = "#{deployment.label.tr('_', '-')}-ose-master#{i}"
```

Comment:
Once we have the ose_master_hosts and ose_node_hosts saved to the DB this should change to loop over the hosts themselves.